sklearn/ensemble/_bagging.py

```diff
@@ -538,6 +564,33 @@ def estimators_samples_(self):
     def n_features_(self):
         return self.n_features_in_
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        base_estimator = self._validate_estimator(self._get_estimator())
```

Comment:
Is there any way that `get_matadata_routing` get called before `fit`?

In such a case, the `base_estimator` instance is potentially not the same as the `base_estimator_` initialized in `fit`. I don't know if it could trigger an issue.

Reply:
I think it's not necessary for the instance to be identical. @adrinjalali can you confirm?

Reply:
Yes, the method can be called before fit. GridSearch would need to do that for instance. That's why we have refactored and added `_validate_estimator` here.

If somebody implements a `get_metadata_routing` which returns a different value if the object is fit vs when it's not, then we can't help them 😁